verification/tagTCI.go

```diff
@@ -3,55 +3,239 @@
 package verification
 
 import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/sha512"
 	"errors"
+	"hash"
+
 	//"reflect"
 	"testing"
 )
 
-func TestTagTCI(d TestDPEInstance, client DPEClient, t *testing.T) {
-	useSimulation := false
-	ctx := getInitialContextHandle(d, client, t, useSimulation)
+const defaultCtxTCITag = TCITag(12345)
+const nonExistentTCITag = TCITag(98765)
+const childCtxTCITag = TCITag(34567)
+
+// Check tagTCI command with default context handle.
+func TestTagTCI(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var err error
+	useSimulation := false // To indicate that simulation context is not used
+
+	// Get default context handle
+	handle := getInitialContextHandle(d, c, t, useSimulation)
 
-	tag := TCITag(12345)
-	// Check to see our tag is not yet found.
-	if _, err := client.GetTaggedTCI(tag); !errors.Is(err, StatusBadTag) {
+	// Check to see our tag is not yet found and then tag default context
+	if _, err := c.GetTaggedTCI(defaultCtxTCITag); !errors.Is(err, StatusBadTag) {
 		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
 	}
 
-	// Tag the default context
-	handle, err := client.TagTCI(ctx, tag)
+	// Tag default context handle and make sure default handle returns
+	// same handle when used for tagging TCI
+	newHandle, err := c.TagTCI(handle, defaultCtxTCITag)
 	if err != nil {
 		t.Fatalf("Could not tag TCI: %v", err)
 	}
+	if *newHandle != *handle {
+		t.Errorf("New context handle from TagTCI was %x, expected %x", newHandle, handle)
+	}
+
+	// Retag a tagged TCI should report error
+	newTag := TCITag(11111)
+	if _, err := c.TagTCI(handle, newTag); !errors.Is(err, StatusBadTag) {
+		t.Fatalf("Re-tagging a tagged TCI returned %v, want %v", err, StatusBadTag)
+	}
+
+	// Fetching a non-existent tag should report error
+	if _, err := c.GetTaggedTCI(TCITag(nonExistentTCITag)); !errors.Is(err, StatusBadTag) {
+		t.Fatalf("GetTaggedTCI returned %v, want %v", err, StatusBadTag)
+	}
+}
+
+// Check whether the ExtendTCI command updates the current TCI and cumulative TCI.
+func TestExtendTCI(d TestDPEInstance, c DPEClient, t *testing.T) {
+	var err error
```

Comment:
Split into new testcase as per review comment.
It extends TCI to existing context and verifies the current/cum measurements.